### DISCUSSION TITLE
Build from source without binpac and bifcl

### DISCUSSION THREAD

**Discussion by ghuband** [zeek/zeek#2537] [Help]
How do I build Zeek from source without binpac and bifcl?  I'm running Zeek on an ARM device.

**Comment by 0xxon** [comment]
Hi,
binpac and bifcl are typically automatically built by Zeek - and required to build Zeek. Could you explain a bit what you are doing/trying to accomplish?

**Comment by ghuband** [comment]
I'm running Zeek on an embedded device and want to minimize what is installed.  My understanding is that binpac and bifcl are not needed for Zeek to run, but are needed to build Zeek.  Ideally, I want to build Zeek with Yocto and not include binpac and bifcl in the binaries.

**Reply by 0xxon** [reply]
binpac and bifcl are already not included in the Zeek binary; they are only used to build the parser/event/bif c++ source files that are compiled into Zeek.

**Reply by ghuband** [reply]
When I build Zeek from source these binaries are installed
$ ls -l /usr/local/zeek/bin
total 307760
-rwxr-xr-x 1 root root    769896 Nov  3 17:30 bifcl
-rwxr-xr-x 1 root root   7063400 Nov  3 17:30 binpac
lrwxrwxrwx 1 root root        32 Oct 27 12:30 bro -> /usr/local/zeek/bin/zeek-wrapper
lrwxrwxrwx 1 root root        32 Oct 27 12:29 bro-config -> /usr/local/zeek/bin/zeek-wrapper
lrwxrwxrwx 1 root root        32 Oct 27 12:30 bro-cut -> /usr/local/zeek/bin/zeek-wrapper
lrwxrwxrwx 1 root root        32 Oct 27 12:30 broctl -> /usr/local/zeek/bin/zeek-wrapper
-rwxr-xr-x 1 root root   6141204 Oct 27 10:12 broker-benchmark
-rwxr-xr-x 1 root root   9175704 Oct 27 10:10 broker-cluster-benchmark
-rwxr-xr-x 1 root root     84437 Sep 19 19:54 btest
-rwxr-xr-x 1 root root      1025 Sep 19 19:54 btest-ask-update
-rwxr-xr-x 1 root root       679 Sep 19 19:54 btest-bg-run
-rwxr-xr-x 1 root root       458 Sep 19 19:54 btest-bg-run-helper
-rwxr-xr-x 1 root root      3046 Sep 19 19:54 btest-bg-wait
-rwxr-xr-x 1 root root      7621 Sep 19 19:54 btest-diff
-rwxr-xr-x 1 root root       398 Sep 19 19:54 btest-diff-rst
-rwxr-xr-x 1 root root       737 Sep 19 19:54 btest-progress
-rwxr-xr-x 1 root root      3423 Sep 19 19:54 btest-rst-cmd
-rwxr-xr-x 1 root root       399 Sep 19 19:54 btest-rst-include
-rwxr-xr-x 1 root root       146 Sep 19 19:54 btest-rst-pipe
-rwxr-xr-x 1 root root       143 Sep 19 19:54 btest-setsid
-rwxr-xr-x 1 root root    128392 Oct 27 05:18 capstats
-rwxr-xr-x 1 root root   1469372 Oct 27 04:57 paraglob-test
-rwxr-xr-x 1 root root     39154 Oct 27 03:59 trace-summary
-rwxr-xr-x 1 root root 289352060 Oct 27 11:13 zeek
-rwxr-xr-x 1 root root    678600 Oct 27 04:56 zeek-archiver
-rwxr-xr-x 1 root root      2354 Oct 27 03:59 zeek-config
-rwxr-xr-x 1 root root     36996 Oct 27 04:56 zeek-cut
-rwxr-xr-x 1 root root       699 Oct 27 03:59 zeek-wrapper
-rwxr-xr-x 1 root root     28656 Oct 27 03:59 zeekctl
-rwxr-xr-x 1 root root     90372 Oct 27 03:59 zkg

What are the minimal executables needed in /usr/local/bin to run Zeek?

**Reply by 0xxon** [reply]
There is no straight answer to that question, because it depends how you want to run Zeek.

Do you just want to run one Zeek process directly from the command line? Then - only the Zeek binary. Do you need zeekctl? Then it gets more complex, because Zeekctl needs zeek, trace-summary, and others. Want zkg to install packages? then you need zkg and the btest bits.

The easiest way to figure out what is needed for what is probably to take a look at our rpm spec-file at https://build.opensuse.org/package/view_file/security:zeek/zeek-nightly/zeek-nightly.spec?expand=1 which splits things into packages.

That being said - Zeek itself will probably be by far the largest executable.

**Reply by ghuband** [reply]
Thanks, that helps.
